commit ec73240b1627cddfd7cef018c7fa1c32e64a721e
Author: Josh Don <joshdon@google.com>
Date:   Tue Aug 4 12:34:13 2020 -0700

    sched/fair: Ignore cache hotness for SMT migration
    
    SMT siblings share caches, so cache hotness should be irrelevant for
    cross-sibling migration.
    
    Signed-off-by: Josh Don <joshdon@google.com>
    Proposed-by: Venkatesh Pallipadi <venki@google.com>
    Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
    Link: https://lkml.kernel.org/r/20200804193413.510651-1-joshdon@google.com

--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -7432,6 +7432,10 @@ static int task_hot(struct task_struct *
 	if (env->sd->flags & SD_SHARE_CPUCAPACITY)
 		return 0;
 
+	/* SMT siblings share cache */
+	if (env->sd->flags & SD_SHARE_CPUCAPACITY)
+		return 0;
+
 	/*
 	 * Buddy candidates are cache hot:
 	 */
